api/http/api_inventory.go

```diff
@@ -416,6 +416,8 @@ func (i *inventoryHandlers) GetDevicesByGroup(w rest.ResponseWriter, r *rest.Req
 	for _, l := range links {
 		w.Header().Add("Link", l)
 	}
+	count, err := i.inventory.GetDeviceCountByGroup(ctx, model.GroupName(group))
+	w.Header().Add("GroupCount", strconv.Itoa(count))
```

Comment:
I understand it's a way of working around an API response change but am not a fan. group count could easily be part of a json response, no reason to stick it in a custom header. of course that would be an api version bump, and would have to be planned, but if we want it - let's do it the proper way.

Reply:
I agree that it would be nice to have this in the response, but thought that it would be tolerable since we have the other pagination information (the page links to lines above) already in the header.
I've seen both ways in other APIs and have no real preference, but know that without this information we struggle in the frontend. So I would vote to keep this in the header as basically a workaround until we have consistent API responses across the endpoints with pagination information.

Reply:
what does "GroupCount" mean here?
does it mean there are _GroupCount_ devices in the given group?
I agree with @mchalski that it would be better to design this first (like create a feature request for the backend or, if you want to do it yourself, create PR with just a doc modification, discuss feature there and then continue with implementation).
If you still want to add a custom header, something like "X-total-count" sounds better to me.

Reply:
yes, GroupCount as in that amount of devices in the requested group and using "X-.." for the custom header should have been obvious to me. Then I will transfer this to it's own ticket and we can continue the discussion there